### ISSUE TITLE
Support endingVersion option in Delta stream source

### ISSUE THREAD

**Issue by tomasbartalos** [delta-io/delta#543]
Since Delta will soon support **startingVersion** and **startingTimestamp** in stream source I would find it useful to have the option of **endingVersion** - similar to endingOffsets in [Kafka streams](https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html).

My use case:
I'd like to downstream all increments from source Delta until the specified commit version. When the commit version is reached, the stream will be terminated.

This feature would give us interesting downstream views like:
- what has been commited to delta yesterday ?
- manual trigger to downstream all commits - this is not the same as `Trigger.Once()` because if the `number_of_commits > maxFilesPerTrigger` only part of the table is streamed.

**Comment by zsxwing** [comment]
Thanks for raising this issue. When the query reaches in `endingVersion`, do you need to stop the query?

**Comment by tomasbartalos** [comment]
Yes, when the `endingVersion` is reached the stream should stop.

**Comment by foss-contributor** [comment]
@tomasbartalos, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
